Sources/HaCWebsiteLib/routes.swift

```diff
@@ -17,7 +17,7 @@ func getWebsiteRouter() -> Router {
     "/git": "https://github.com/hackersatcambridge/git-workshop-2017"
   ]))
 
-  router.all("/static", middleware: StaticFileServer(path: "./static/dist"))
+  Assets.addTo(router: router)
```

Comment:
Don't love that we're obscuring the fact that we're adding middleware here. Is there any benefit to having an `addTo` function? Will this ever be used more than once in a given codebase?

Reply:
This was so we can abstract the fact that there is this `/static` route, and that `Assets` can own it completely. Developers don't need to know that this thing exists

Reply:
Why is obscuring adding to middleware necessarily a bad thing?

Reply:
I agree with @jaredkhan, it does make it seem a bit 'magic'.

However, looking at it again through the idiomatic Swift lens, it does make more sense. I don't think it's too bad.

Reply:
When adding things to the router, order matters. If we for (a contrived) example had a workshop on static analysis and tried to short link it to `/static`, we might skim the routes file to find out why this isn't working. Hiding this in Assets would make figuring this out harder.

Reply:
@jaredkhan ideally we won't have `/static` hardcoded in even more places in our codebase. We could make the function name more informative?

`Assets.addFileServingMiddleware(to: router)`

Reply:
@Pinpickle I would be happy with that
I was trying to think of funciont names to help with this, but came up blank. That looks good.

Reply:
@pinpickle my point was that middleware can basically hijack whatever is fed to it. I think hiding the fact that `/static` is in use for static assets is confusing

Reply:
@jaredkhan What if we explicitly passed `/static` to the `Assets` enum? What if it was a struct that was explicitly instantiated once, instead of an enum with global stuff going on?

Reply:
I think we should just pull this function outside of Assets. I disagree with your point @Pinpickle that 'Developers don't need to know that this thing exists'. It affects the routing behaviour on a path that they _might_ use so they _do_ need to know it exists.

Reply:
Okay I think I know how to go ahead and make this work for everyone, watch this space